src/main/kotlin/com/icat/orboarding/user/adapters/outbound/persistence/mysql/UserPersistence.kt

```diff
@@ -34,4 +34,7 @@ class UserPersistence(private val userRepository: UserRepository) : UserPersiste
 
     private fun UserDomain.toUserEntity() =
         UserEntity(id = id!!, email = email, password = password!!)
+
+    override fun getUserByEmail(email: String): UserEntity =
```

Comment:
```suggestion
    override fun getUserByEmail(email: String): UserDomain =
```
Opa Maninho, o correto aqui é sempre externalizarmos as classes do contexto de adapter para a camada de application, pois se não, acabamos ferindo o contexto da camada de application conhecer somente as classes que são de application.

Reply:
Boa, saquei!!!

Lá no AuthenticationService, no método ```loadUserByUsername```, é necessário uma classe que implemente a interface ```UserDetails```, que eu acabei deixando na UserEntity com base nos exemplos que vi. Não tinha me atentado a essa questão da quebra da arquitetura mesmo.

Pensei em criar uma outra classe, UserSecurity, que implementaria ela e faria esse método da persistence retornar ela usando aquelas funções de extensão, tipo um `UserEntity.toUserSecurity()`. Ou então fazer a `UserDomain` implementar essa `UserDetails`, mas n me parece certo deixar essa questão de sec numa classe dentro de application.

Reply:
Vamos deixar assim por enquanto maninho, mas vou criar uma terefinha de débito técnico pra gente tentar encontrar uma forma de contornar isso, o que acha?